### PR TITLE
[Merged by Bors] - feat(Order): Galois connection between `pred` and `succ`

### PR DESCRIPTION
**Pull request by YaelDillies**
This was a longstanding TODO. Kudos to Kevin for putting the pieces together!

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Order/SuccPred/Basic.lean [hunk 1]
```diff
@@ -445,15 +451,11 @@ theorem le_le_succ_iff : a ≤ b ∧ b ≤ succ a ↔ b = a ∨ b = succ a := by
   · exact ⟨le_succ a, le_rfl⟩
 #align order.le_le_succ_iff Order.le_le_succ_iff
 
-theorem _root_.CovBy.succ_eq (h : a ⋖ b) : succ a = b :=
-  (succ_le_of_lt h.lt).eq_of_not_lt fun h' => h.2 (lt_succ_of_not_isMax h.lt.not_isMax) h'
-#align covby.succ_eq CovBy.succ_eq
+/-- See also `Order.le_succ_of_wcovBy`. -/
+lemma succ_eq_of_wcovBy (h : a ⋖ b) : succ a = b := (succ_le_of_lt h.lt).antisymm h.wcovBy.le_succ
```
Comment (Ruben-VandeVelde) [inline Mathlib/Order/SuccPred/Basic.lean:455]: Shouldn't this be `of_covBy`?

Comment (YaelDillies) [reply Mathlib/Order/SuccPred/Basic.lean:455]: I'm not on my computer (for a while) so you could please fix it yourself/suggest the change in a comment?

**Comment by semorrison** [comment]
Only hesitation is the import increase, otherwise LGTM. Happy if someone wants to merge regardless, or to ask for a split.

**Comment by YaelDillies** [comment]
As the author of `Order.SuccPred.Basic`, I claim this import increase is fine 😄

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**